frontend/src/utils/StringFormatting.tsx

```diff
@@ -2,8 +2,13 @@ import { format } from 'date-fns'
 
 const millisecondsInADay = 8.64e7
 
-export const convertUTCDateToLocalDate = (date: Date): Date =>
-    new Date(date.getTime() - date.getTimezoneOffset() * 60 * 1000)
+export const convertUTCDateToLocalDate = (date: Date): Date => {
+    // If lastChar is Z, typescript assumes the date to be UTC
+    const lastChar = date.toString().slice(-1)
+    if (lastChar === 'Z') return new Date(date)
```

Comment:
I'm confused, this function is called convert from UTC to Local but if we get a date in UTC we don't do anything and if we receive one in local time then we change it. Shouldn't it then be change local time to UTC ?

Reply:
We always get the date as UTC. If there is a Z typescript assumes (correctly) that the time is UTC and will handle the conversion to local time. If not it will assume that the date already is local time and we need to remove the offset to get the correct local time

Reply:
I thought this could be the case, a bit confusing with the naming but can't really think of a better way to make it more explicit. Whoever looks at it will have to dig in the code